activerecord/lib/active_record/validations.rb

```diff
@@ -63,12 +63,17 @@ def valid?(context = nil)
 
   protected
 
+    def run_validations? # :nodoc:
+      @run_validations
+    end
+
     def raise_validation_error
       raise(RecordInvalid.new(self))
     end
 
     def perform_validations(options={}) # :nodoc:
-      options[:validate] == false || valid?(options[:context])
+      @run_validations = options[:validate] != false
```

Comment:
This ivar seems.. not so great: it appears to be "cheating" to get the value through several layers of method call.

Like @rafaelfranca I'm not sure whether we want to do this... but I think it'd be easier to reason about, as a change that actually pushes it down through all the methods.